src/test/java/io/airlift/compress/thirdparty/HadoopLzoCompressor.java

```diff
@@ -13,30 +13,29 @@
  */
 package io.airlift.compress.thirdparty;
 
-import com.hadoop.compression.lzo.LzoCodec;
 import io.airlift.compress.Compressor;
 import io.airlift.compress.HadoopNative;
-import org.apache.hadoop.conf.Configuration;
+import org.anarres.lzo.hadoop.codec.LzoCompressor;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 public class HadoopLzoCompressor
-    implements Compressor
+        implements Compressor
 {
     static {
         HadoopNative.requireHadoopNative();
     }
 
-    private static final Configuration HADOOP_CONF = new Configuration();
-
+    private static final int maxOutputBufferSize = 128 * 1024 * 1024;
     private final org.apache.hadoop.io.compress.Compressor compressor;
 
     public HadoopLzoCompressor()
     {
-        LzoCodec codec = new LzoCodec();
-        codec.setConf(HADOOP_CONF);
-        compressor = codec.createCompressor();
+        // use a large enough buffer to avoid framing
```

Comment:
Does this mean that our implementation does not support Hadoop framing?  I assume that might be a problem for some users.

Reply:
Some compression algorithms, like snappy, have internal framing which is used when the data presented to the compressor is too big to do in one shot.

I'm fine merging the current fix and we can look at this in another PR.  Does that sound reasonable to you?

Reply:
Looks good to me. Please merge this fix.

But I think we still need a clear definition of the scope of `LzoCompressor` / `LzoDecompressor` and have it documented somewhere in the code.